### ISSUE TITLE
Docker for Mac cannot initiate properly

### ISSUE THREAD

**Issue by secjex** [docker/for-mac#119]
### Expected behavior
it should be in running status
### Actual behavior
Always in starting status, quit and reopen not working
### Information

Diagnostic ID: D0A28EE4-7886-49BA-8BAC-CA6BA2ABCA07
Docker for Mac: 1.12.0-beta21 (Build 11019)
macOS: Version 10.11.6 (Build 15G31)
[ERROR]  docker-cli
         docker ps failed
[OK]     app
[OK]     moby-syslog
[OK]     disk
[OK]     virtualization
[OK]     system
[OK]     menubar
[OK]     osxfs
[OK]     db
[OK]     slirp
[OK]     moby-console
[OK]     logs
[OK]     vmnetd
[OK]     env
[OK]     moby
[OK]     driver.amd64-linux

### Steps to reproduce

after upgrade to this version

**Comment by mchiang0610** [comment]
#574 may be related... 

```
Information

Diagnostic ID: DCE04336-8053-4026-87FD-CBFB6D23258E
Docker for Mac: 1.12.0-a (Build 11213)
macOS: Version 10.11.6 (Build 15G1004)
[ERROR] docker-cli
docker ps failed
[OK] app
[OK] moby-syslog
[OK] disk
[OK] virtualization
[OK] system
[OK] menubar
[OK] osxfs
[OK] db
[OK] slirp
[OK] moby-console
[OK] logs
[OK] vmnetd
[OK] env
[ERROR] moby
/Users/zardoz/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/console-ring does not exist
[ERROR] driver.amd64-linux
Docker.app/Contents/MacOS/com.docker.driver.amd64-linux -db is not running
```

**Comment by mchiang0610** [comment]
#551 may be related -- with disk check failure as well. 

```
Diagnostic ID: 7DB0C3CB-5505-425C-9AC6-D67C93E29D78
Docker for Mac: 1.12.0-a (Build 11213)
macOS: Version 10.11.6 (Build 15G1004)
[ERROR] docker-cli
docker ps failed
[OK] app
[OK] moby-syslog
[ERROR] disk
disk check failed with: Failure("exec: /Applications/Docker.app/Contents/Resources/bin/../../../Contents/MacOS/qemu-img check /Users/debersom/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/Docker.qcow2 > /tmp/7DB0C3CB-5505-425C-9AC6-D67C93E29D78/20160908-105639/qemu-img-check.stdout 2> /tmp/7DB0C3CB-5505-425C-9AC6-D67C93E29D78/20160908-105639/qemu-img-check.stderr: exit 2")
[OK] virtualization
[OK] system
[OK] menubar
[OK] osxfs
[OK] db
[OK] slirp
[OK] moby-console
[OK] logs
[OK] vmnetd
[OK] env
[OK] moby
[OK] driver.amd64-linux
```

**Comment by mchiang0610** [comment]
From #564 

```
Diagnostic ID: E3285C35-A6EC-4EBA-B9B3-6CD6D8803BF7
Docker for Mac: 1.12.0-a (Build 11213)
macOS: Version 10.11.1 (Build 15B42)
[ERROR] docker-cli
docker ps failed
[OK] app
[OK] moby-syslog
[ERROR] disk
disk check failed with: Failure("exec: /Applications/Docker.app/Contents/Resources/bin/../../../Contents/MacOS/qemu-img check /Users/willcheng/Library/Containers/com.docker.docker/Data/com.docker.driver.amd64-linux/Docker.qcow2 > /tmp/E3285C35-A6EC-4EBA-B9B3-6CD6D8803BF7/20160909-144611/qemu-img-check.stdout 2> /tmp/E3285C35-A6EC-4EBA-B9B3-6CD6D8803BF7/20160909-144611/qemu-img-check.stderr: exit 2")
[OK] virtualization
[OK] system
[OK] menubar
[OK] osxfs
[OK] db
[OK] slirp
[OK] moby-console
[OK] logs
[OK] vmnetd
[OK] env
[OK] moby
[OK] driver.amd64-linux
```

**Comment by aripalo** [comment]
This has been already [mentioned](https://github.com/docker/for-mac/issues/398#issuecomment-241587291) [times](https://github.com/docker/for-mac/issues/367#issuecomment-241559658) [several](https://github.com/docker/for-mac/issues/350#issuecomment-241558668) [times](https://github.com/docker/compose/issues/3904#issuecomment-246632820) in other issues (closed in favour of this issue), but for anyone ending here and not knowing what to do:

A _temporary “solution”_ (until an actual fix comes along) is to run `sudo pkill docker` and then restart the Docker for Mac app.

**Comment by homofortis** [comment]
Make sure you have Intel Virtualization enabled in your BIOS. In my case, docker started after enabling it.

**Comment by secjex** [comment]
after I click the `reset to factory defaults`, it works fine.

**Comment by bschalme** [comment]
Neither `sudo pkill docker` nor clicking `reset to factory defaults` worked for me. After trying both, Docker 1.12.2-rc1-beta27 still remains in starting mode on my MacBook Pro, it never finishes starting.

**Comment by samoht** [comment]
The cause of the initial issue is AUFS corruption:

```
time="2016-08-03T01:18:28.774255852Z" level=debug msg="Using default logging driver json-file"
time="2016-08-03T01:18:28.774729056Z" level=debug msg="Golang's threads limit set to 13950"
time="2016-08-03T01:18:28.775118420Z" level=debug msg="[graphdriver] trying provided driver \"aufs\""
time="2016-08-03T01:18:28.780220435Z" level=debug msg="Using graph driver aufs"
time="2016-08-03T01:18:28.783331024Z" level=debug msg="Cleaning up old mountid : start."
time="2016-08-03T01:18:28.783662757Z" level=fatal msg="Error starting daemon: mkdir /var/lib/docker/image: structure needs cleaning"
time="2016-08-04T01:50:11Z" level=warning msg="Running experimental build"
time="2016-08-04T01:50:11.871847300Z" level=debug msg="docker group found. gid: 50"
time="2016-08-04T01:50:11.873715194Z" level=debug msg="Listener created for HTTP on unix (/var/run/docker.sock)"
time="2016-08-04T01:50:11.875302274Z" level=info msg="libcontainerd: new containerd process, pid: 1349"
time="2016-08-04T01:50:11.876495966Z" level=debug msg="libcontainerd: containerd connection state change: TRANSIENT_FAILURE"
```

Closing the issue as this should now be safer in more recent betas.

**Comment by samoht** [comment]
Also I went through all the referenced issues and tried to triage them a bit more precisely. Lots of docker/hyperkit#50, lots of partial installations, lots of AUFS corruption issues and few osx invariant breakage. Some of these will be fixed in Beta 28, and we are working in fixing the other ones.

**Comment by bastien** [comment]
I had the same problem, running Docker for Mac 1.12.3 (13776)
My mac quit unexpectedly and after the reboot `docker ps` no longer worked. `sudo pkill docker` didn't solve the problem, I had to reset to factory settings and fetch all my images again.

**Comment by micron** [comment]
Same here on El Capitan (10.11.6) with Docker for Mac 1.13.1 (15353). Reseting to factory defaults resolves this temporary.

**Comment by florrain** [comment]
I ran a SQL import overnight to a mySQL container in docker-machine (latest docker for mac Version 17.03.1-ce-mac5 (16048)) and when I wake up and check the status of the import, it's stuck halfway through. `docker` is unresponsive, containers don't respond and trying to restart via the GUI never finishes.

I first tried native docker and couldn't get the import working and thought it was because of the qcow max size of 65gb and now I'm seeing this issue. I'm starting to wonder if it's the same after all. 

Basically a huge SQL import or long process working in the background while the laptop is idle make docker freeze. And that's a huge pain in the ass for me, probably for somebody else out there. Is there a bug filed for this or a known workaround?

I'm on macOS Sierra 10.12.4 + MBP 2016.

> Neither sudo pkill docker nor clicking reset to factory defaults worked for me. After trying both, Docker 1.12.2-rc1-beta27 still remains in starting mode on my MacBook Pro, it never finishes starting.

Same here for me.

**Comment by rbflow** [comment]
facing the same issue for the latest one I downloaded and installed last night. Can anyone help?

**Comment by marnen** [comment]
@samoht Looks like there have been at least 4 reports of this persisting since you closed the issue. Should this be reopened?

**Comment by marnen** [comment]
@akimd Then I'm sorry if I spoke too soon. From a user's perspective, it sure looks like a similar issue, and I'll admit to getting a bit frustrated by seeing lots of reports and little resolution...

**Comment by akimd** [comment]
Nah, don't worry.  But be sure there are _tons_ of different reasons that lead to the same symptom: "it does not work" :)
